PSReadLine/Prediction.cs

```diff
@@ -45,13 +45,17 @@ public static void AcceptSuggestion(ConsoleKeyInfo? key = null, object arg = nul
             Prediction prediction = _singleton._prediction;
             if (prediction.ActiveView is PredictionInlineView inlineView && inlineView.HasActiveSuggestion)
             {
+                var start = inlineView.InputText.Length;
+
                 // Ignore the visual selection.
                 _singleton._visualSelectionCommandCount = 0;
 
                 inlineView.OnSuggestionAccepted();
 
                 using var _ = prediction.DisableScoped();
-                Replace(0, _singleton._buffer.Length, inlineView.SuggestionText);
+
+                _singleton._current = start;
+                Insert(inlineView.SuggestionText.Substring(_singleton._current));
```

Comment:
As alluded to in the PR description, I also slightly changed the `AcceptSuggestion` and `AcceptNextSuggestionWord` methods which were doing a replacement of the buffer. This PR now makes those methods add the appropriate portion of text so that undoing the operation behaves more inline with most users’ expectations. Externally, the behaviour of those methods has not changed.

It simplifies the code and has the same external behaviour while not using a delete operation at all. This therefore makes it invariant in the face of whether to customize the undo mechanism with `EditItemDelete` or not in other parts of the code.